### PR TITLE
Rewrite StartsWith and handle CustomBufferedStream Async for .net standard2.0

### PR DESCRIPTION
**Pull request by StephaneGraziano**
Doneness:
- [X] Build is okay - I made sure that this change is building successfully.
- [X] No Bugs - I made sure that this change is working properly as expected. It doesn't have any bugs that you are aware of. 
- [X] Branching - If this is not a hotfix, I am making this request against the master branch 

StartsWith could be rewritten with the new PeekBytesAsync instead of looping.
(An other optimisation could be remove the Http Method check because we don't use it anywhere in TWP)

### CODE REVIEW

File: src/Titanium.Web.Proxy/Helpers/HttpHelper.cs [hunk 1]
```diff
@@ -143,37 +145,23 @@ internal static Task<int> IsPriMethod(ICustomStreamReader clientStreamReader)
...
-                {
-                    return isExpected ? 1 : 0;
-                }
 
-                char ch = (char)b;
-                if (!char.IsLetter(ch))
-                {
-                    return -1;
-                }
+            var vBuffer = await clientStreamReader.PeekBytesAsync(0, lengthToCheck, cancellationToken);
```
Comment (honfika) [inline src/Titanium.Web.Proxy/Helpers/HttpHelper.cs]: I not really like this refactoring. Why is it needed?
Is any bug fixed in this change?

My problem is that it causes multipe new object allocations:
PeekBytesAsync : new byte[]
GetString: new string
Regex.. at least it shoudl be compiled, but why, when we just want to check only english A-Z characters?

Comment (honfika) [reply src/Titanium.Web.Proxy/Helpers/HttpHelper.cs]: IS the multipel PeekBytesAsync slow? In that case maybe there should be an overload for PeekBytesAsync, which accepts a buffer, and fills that. Like the Read methods.

Comment (StephaneGraziano) [reply src/Titanium.Web.Proxy/Helpers/HttpHelper.cs]: I think await state-machine on loop is inefficient.
I'm of course ok for passing a buffer (we could use IBufferPool then).
The initial rule is (3 A-Z + space) but as I said, we never use the fact the method is HTTP, we could only write this :


    int iRet = -1;
    const int lengthToCheck = 10;
    byte[] buffer = null;
    try
    {
        buffer = bufferPool.GetBuffer(lengthToCheck);
        int iPeeked = await clientStreamReader.PeekBytesAsync(buffer, 0, lengthToCheck, cancellationToken);
        if (iPeeked > 0)
        {
            var httpMethod = defaultEncoding.GetString(buffer, 0, iPeeked);
            iRet = httpMethod.StartsWith(expectedStart) ? 1 : -1;
        }
    }
    finally
    {
        bufferPool.ReturnBuffer(buffer);
    }
    return iRet;
    //and why not give up the int ret -1/1 to bool

Comment (StephaneGraziano) [reply src/Titanium.Web.Proxy/Helpers/HttpHelper.cs]: I rewrote to have the best of the 2 worlds : use BufferPool with no string alloc and of course without looping on the await PeekByte. Let me know if it's ok for you.

**Comment by honfika** [comment]
For me it looks good.

**Pull request merged**